Mathlib/RingTheory/LinearDisjoint.lean

```diff
@@ -412,6 +470,235 @@ variable {A B}
+theorem of_isField' {A : Type v} [CommRing A] {B : Type w} [CommRing B]
+    [Algebra R A] [Algebra R B] (H : IsField (A ⊗[R] B))
+    (fa : A →ₐ[R] S) (fb : B →ₐ[R] S) (hfa : Function.Injective fa) (hfb : Function.Injective fb) :
+    fa.range.LinearDisjoint fb.range := by
+  apply of_isField
+  exact Algebra.TensorProduct.congr (AlgEquiv.ofInjective fa hfa)
+    (AlgEquiv.ofInjective fb hfb) |>.symm.toMulEquiv.isField _ H
+
+-- TODO: move to suitable place
+variable (R) in
+/-- If `M`, `N` are `R`-modules, there exists an injective `R`-linear map from `R` to `N`,
+and `M` is a nontrivial flat `R`-module, then `M ⊗[R] N` is nontrivial. -/
+theorem _root_.TensorProduct.nontrivial_of_linearMap_injective_of_flat_left
+    (M N : Type*) [AddCommGroup M] [AddCommGroup N] [Module R M] [Module R N]
+    (f : R →ₗ[R] N) (hf : Function.Injective f) [Module.Flat R M] [Nontrivial M] :
+    Nontrivial (M ⊗[R] N) :=
+  Module.Flat.lTensor_preserves_injective_linearMap (M := M) f hf |>.comp
+    (TensorProduct.rid R M).symm.injective |>.nontrivial
+
+-- TODO: move to suitable place
+variable (R) in
+/-- If `M`, `N` are `R`-modules, there exists an injective `R`-linear map from `R` to `M`,
+and `N` is a nontrivial flat `R`-module, then `M ⊗[R] N` is nontrivial. -/
+theorem _root_.TensorProduct.nontrivial_of_linearMap_injective_of_flat_right
+    (M N : Type*) [AddCommGroup M] [AddCommGroup N] [Module R M] [Module R N]
+    (f : R →ₗ[R] M) (hf : Function.Injective f) [Module.Flat R N] [Nontrivial N] :
+    Nontrivial (M ⊗[R] N) :=
+  Module.Flat.rTensor_preserves_injective_linearMap (M := N) f hf |>.comp
+    (TensorProduct.lid R N).symm.injective |>.nontrivial
+
+-- TODO: move to suitable place
+variable (R) in
+/-- If `A`, `B` are `R`-algebras, `R` injects into `B`,
+and `A` is a nontrivial flat `R`-algebra, then `A ⊗[R] B` is nontrivial. -/
+theorem _root_.Algebra.TensorProduct.nontrivial_of_algebraMap_injective_of_flat_left
+    (A B : Type*) [CommRing A] [CommRing B] [Algebra R A] [Algebra R B]
+    (h : Function.Injective (algebraMap R B)) [Module.Flat R A] [Nontrivial A] :
+    Nontrivial (A ⊗[R] B) :=
+  TensorProduct.nontrivial_of_linearMap_injective_of_flat_left R A B (Algebra.linearMap R B) h
+
+-- TODO: move to suitable place
+variable (R) in
+/-- If `A`, `B` are `R`-algebras, `R` injects into `A`,
+and `B` is a nontrivial flat `R`-algebra, then `A ⊗[R] B` is nontrivial. -/
+theorem _root_.Algebra.TensorProduct.nontrivial_of_algebraMap_injective_of_flat_right
+    (A B : Type*) [CommRing A] [CommRing B] [Algebra R A] [Algebra R B]
+    (h : Function.Injective (algebraMap R A)) [Module.Flat R B] [Nontrivial B] :
+    Nontrivial (A ⊗[R] B) :=
+  TensorProduct.nontrivial_of_linearMap_injective_of_flat_right R A B (Algebra.linearMap R A) h
+
+-- TODO: move to suitable place
+/-- If `A`, `B` are nontrivial algebras over a field `F`, then `A ⊗[F] B` is nontrivial. -/
+theorem _root_.Algebra.TensorProduct.nontrivial_of_field
```

Comment:
Where should this go? It uses `Module.Flat` and `Module.Free.of_divisionRing`. `#find_home` suggests `[Mathlib.RingTheory.LocalRing.Module, Mathlib.RingTheory.LinearDisjoint, Mathlib.Algebra.Category.Ring.LinearAlgebra, Mathlib.RingTheory.Regular.RegularSequence, Mathlib.RingTheory.Unramified.Finite]` (which includes this file), but none of them seems optimal. What about creating a new file `Mathlib.RingTheory.TensorProduct.Nontrivial`?

Reply:
I guess this is resolved by [this comment](https://github.com/leanprover-community/mathlib4/pull/19832/files#r1875972697) :)

Reply:
Yes but `nontrivial_of_algebraMap_injective_of_isDomain` should go that file anyway.